game/game.py

```diff
@@ -369,12 +372,18 @@ def board(self):
 
     def action_manager(self, action, from_row, from_col, direction):
         to_row, to_col = translate_position(from_row, from_col, direction)
-        if action == MOVE:
-            self.is_valid_move(
-                from_row,
-                from_col,
-                to_row,
-                to_col,
-                self.current_player)
-        elif action == SHOOT:
-            self.shoot_arrow(from_row, from_col, direction)
+
+        try:
+            if action == MOVE:
+                self.is_valid_move(
+                    from_row,
+                    from_col,
+                    to_row,
+                    to_col,
+                    self.current_player)
+
```

Comment:
In this line we should  turn the invalid move counter into `0`
```suggestion
self.current_player.invalid_moves_count = 0
```

Reply:
Thanks, now I'll add it, I'm going to make a commit amend and then I'll upload the PR again.